crates/proto/src/lib.rs

```diff
@@ -115,14 +115,20 @@ pub mod iocompat {
 // This trait is created to facilitate running the tests defined in the tests mod using different types of
 // executors. It's used in Fuchsia OS, please be mindful when update it.
 pub trait Executor {
+    /// Create the implementor itself.
+    fn new() -> Self;
```

Comment:
FYI, this will make the type not object safe. This might be ok, but it's something to be aware of.